aea/aea_builder.py

```diff
@@ -590,6 +604,8 @@ def build(self, connection_ids: Optional[Collection[PublicId]] = None) -> AEA:
         self._load_and_add_protocols()
         self._load_and_add_contracts()
         connections = self._load_connections(identity.address, connection_ids)
+        # add alrady laoded connections
+        connections += self._connections
```

Comment:
what if a connection is in `self._connections` and also loaded above?

Reply:
the main idea that the developer is responsible for manual construction of the AEA.
so he had to control everything he does. and this also requires better configuration provided for components

Reply:
I don't think putting this responsibility on the developer will work. We should add consistency checks.